pkg/jobs/jobsprofiler/profiler.go

```diff
@@ -46,6 +48,22 @@ func StorePlanDiagram(
 
 			dspKey := profilerconstants.MakeDSPDiagramInfoKey(timeutil.Now().UnixNano())
 			infoStorage := jobs.InfoStorageForJob(txn, jobID)
+
+			// Limit total retained diagrams by culling older ones as needed.
+			count, err := infoStorage.Count(ctx, profilerconstants.DSPDiagramInfoKeyPrefix, profilerconstants.DSPDiagramInfoKeyMax)
+			if err != nil {
+				return err
+			}
+			const keep = MaxRetainedDSPDiagrams - 1
+			if toCull := count - keep; toCull > 0 {
```

Comment:
I'm wondering if we want to cap this to some reasonable upper bound of how many items we might want to delete in a single transaction?

Reply:
I limited it to 100 for now since why not, but I still think that if we're worried about this one, we should be more worried about `DELETE FROM system.job_info WHERE job_id = ANY($1)` in `cleanupOldJobsPage`.